lib/controller.js

```diff
@@ -170,6 +177,134 @@ Tessel.get = function(opts) {
+
+    if (opts.timeout && typeof opts.timeout === 'number') {
+      // Stop the search after that duration
+      setTimeout(stopSeeker, opts.timeout * 1000);
+    } else {
+      // default to 3 seconds searching
+      setTimeout(stopSeeker, 3000);
+    }
+    logs.info('Searching accessable Tessels ...');
+
+    var seeker = new discover.TesselSeeker().start();
+    var tessels = [];
+
+    // When we find Tessels
+    seeker.on('tessel', function(tessel) {
+      controller.closeTesselConnections(tessel)
+        .then(function() {
+          var known = false;
+          if (tessels.length >= 1) {
+            for (var i in tessels) {
+              if (tessel.connections[0].ip === tessels[i].connections[0].ip && tessel.connections[0].connectionType === 'LAN') {
+                // Your network topology causes a single tessel is found twice
+                debug('known = true');
+                known = true;
+              }
+            }
+
+          }
+          if (known) {
+            debug('Due to your Network-Topology ' + tessel.name + ' is found twice! (' + tessel.connections[0].ip + ')');
+          } else {
+            // Add it to our array
+            tessels.push(tessel);
+          }
+        });
+    });
+
+    seeker.on('error', function(e) {
+      reject(e);
+    });
+
+    function stopSeeker() {
+      try {
+        // If there were no Tessels found
+        if (!tessels || tessels.length === 0) {
+          // Report the sadness
+          debug('We are searching the following networks:\n', networkInterfaces);
+          debug('\n(Important: Check your Netmask - we do not follow gateways for discovering!)');
+          logs.warn('No Tessels found (DEBUG=controller t2 root # might help!)');
+          resolve();
+        }
+        seeker.stop();
+        seeker = null; // preventing memory leaks
+        return resolve(tessels);
+      } catch (e) {
+        reject(e);
+      }
+    }
+  });
+};
+/*
+  The T2 root command is used to login into the firmware and gaining superuser access.
+  The security is provided by RSA - your Tessel get the keys while provisioning via USB.
+  
+  If you have only one Tessel, the T2 root command will login directly else there is 
+  a ncurses like menu you can select the Tessel you like to gain root access...
+
+  The structure of code and maybe unusual parts are paid due being testable.
+  The lightweight terminal-menu isn't written testable what needs a little bit hacking.
+
+  Finally some parts are causing from my personnal learning curve about writing unit tests using sinon!
+*/
+controller.root = function(opts) {
+  // ~ conversion to home because spawn isn't able to handle this right
+  if (opts.path && opts.path.substring(0, 1) === '~') {
+    var home = process.env.HOME;
+    opts.path = opts.path.replace('~', home);
+  }
+  var rtm;
+  if (!opts.menu) {
+    // TODO: Tessel cooperate identity conform menu required (color, ascii logo)
+    rtm = Menu({
```

Comment:
OK like current idea of [logSelected](https://github.com/Student007/t2-cli/blob/menu/lib/controller.js#L82-L114) but moved to [logAndFinish else {...}](https://github.com/tessel/t2-cli/blob/master/lib/controller.js#L162-L168) (btw: the [new Menu](https://github.com/Student007/t2-cli/blob/menu/lib/controller.js#L92-L98) is moved to [prepareMenu](https://github.com/Student007/t2-cli/blob/menu/lib/controller.js#L247))

> @Student007 I don't recommend storing state in a file across CLI calls because things can get hairy if you run two commands at once or another program is using the CLI API. In the latter case, you could have two processes writing to the file at the same time. 

You are right. But in this special case it overrides the file when the user is doing interaction. Another process what overrides the file would fail the same way two terminals try to pipe data into one file or do other stuff with the same target.

I think it is the only useful way to fix a special Tessel to work with. I think there would be cases a developer will have e.g. 8 Tessels with different apps what should work together and he likes to select the current one without typing every time. He would use `t2 list` and select one. 

Another idea would be a new command `t2 select` what will show the menu to fix the Tessel to work with. Because it isn't possible to `export process.env.TESSEL` outside the running process, the only idea is to use a `~/.tesse./selectedTessel.log` (with the risk of other developers will hack around: My current non published version is checking whether the written name is matching an existing Tessel).